ledger/participant-integration-api/src/main/scala/platform/store/cache/MutableCacheBackedContractStore.scala

```diff
@@ -115,9 +115,9 @@ private[platform] class MutableCacheBackedContractStore(
   )(implicit
       loggingContext: LoggingContext
   ): Either[Set[ContractId], (Set[Timestamp], Set[ContractId])] = {
-    val cacheQueried = ids.map(id => id -> contractsCache.get(id))
+    val cacheQueried = ids.view.map(id => id -> contractsCache.get(id)).toVector
```

Comment:
This looks good, but we do not have any test for this :-(

Reply:
Frankly, I'm actually surprised that using `view` avoids building the intermediate set entirely (and therefore removing the uniqueness enforcement of the elements):
```
scala> Set(1,2,3).view.map(_ => 1).toVector
val res0: scala.collection.immutable.Vector[Int] = Vector(1, 1, 1)

scala> Set(1,2,3).map(_ => 1).toVector
val res1: scala.collection.immutable.Vector[Int] = Vector(1)
```

Reply:
@mziolekda We have unit tests asserting this (see https://github.com/digital-asset/daml/blob/main/ledger/participant-integration-api/src/test/suite/scala/platform/store/cache/MutableCacheBackedContractStoreSpec.scala#L333-L366) Or did you meant something else?

Reply:
> Frankly, I'm actually surprised that using view avoids building the intermediate set entirely (and therefore removing the uniqueness enforcement of the elements):

Yes, the view is a safe wrapper around an iterators.